model/user/db.go

```diff
@@ -336,9 +341,55 @@ func GetOrCreateUser(userId, displayName, email, accessToken, refreshToken strin
 		return nil, errors.Wrapf(err, "decoding user '%s'", userId)
 
 	}
+
+	if err := setSlackInformation(env, u); err != nil {
+		return nil, errors.Wrapf(err, "setting slack information for user '%s'", userId)
+	}
+
 	return u, nil
 }
 
+func setSlackInformation(env evergreen.Environment, u *DBUser) error {
+	if u.Settings.SlackMemberId != "" {
+		// user already has a slack member id set
+		return nil
+	}
+	if u.EmailAddress == "" {
+		// we can't fetch the slack information without the user's email address
+		return errors.New("user has no email address")
+	}
+
+	// get the slack userID and memberId from the user's email
+	slackEnv := env.Settings().Slack
+	slackUser, err := slackEnv.Options.GetSlackUser(slackEnv.Token, u.EmailAddress)
+	if err != nil {
+		return errors.Wrapf(err, "getting Slack user with email address '%s'", u.EmailAddress)
+	}
+	if slackUser == nil {
+		grip.Error(message.Fields{
+			"message": fmt.Sprintf("Couldn't find slack user by email address", u.EmailAddress),
```

Comment:
Can we include the email address as a separate field?

Reply:
done